### ISSUE TITLE
Returning Web Response in Azure SWA is not working

### ISSUE THREAD

**Issue by targetlucked69** [nitrojs/nitro#2590]
### Environment

Nuxt: 3.12.3
Node: 20

### Reproduction

Deploy a basic Nuxt app with an api that returns a web response like

```ts
export default function eventHandler((event) => {
  return new Response(JSON.stringify({ username: 'test' }), { headers: { 'Content-Type': 'application/json' } })
})
```

### Describe the bug

Instead of the actual json, Im getting an empty object returned when deployed to Azure SWA:

```json
// /api/user
{}
```

**Comment by itpropro** [comment]
I was able to reproduce this. But there are alternative options to construct and send a response that work with SWA and are more the "Nitro like":

Directly returning the object (Nitro automatically sets the headers):

```
export default defineEventHandler(() => {
  return { username: 'test' }
})
```

Using the send function:

```
export default defineEventHandler((event) => {
  return send(event, { username: 'test' }, 'application/json')
  // also possible
  // return send(event, JSON.stringify({ username: 'test' }), 'application/json')
})
```

Both are tested and confirm to work with Static Web Apps.

Do you want to keep this issue open @pi0 or would you agree not to use the native Response object and instead use the Nitro utilities?

**Comment by itpropro** [comment]
> Returning a new `Response` should be supported in handlers. We should investigate why [`body`](https://github.com/unjs/nitro/blob/v2/src/presets/azure/runtime/azure-swa.ts#L39) (which can be a stream) is not handled, it is a bug.

I verified that this is also a bug in the `azure-function` preset.
It is definitely cause by Nitro handling the response binding for Azure Functions

```
  context.res = {
    status,
    cookies: getAzureParsedCookiesFromHeaders(headers),
    headers: normalizeLambdaOutgoingHeaders(headers, true),
    body,
  };
```

We should address that with the update to Azure Functions runtime 4.

**Comment by pi0** [comment]
thanks for investigation @itpropro. Does runtime v4 support accept `ReadableStream` as valid `body`?

**Comment by itpropro** [comment]
> thanks for investigation @itpropro. Does runtime v4 support accept `ReadableStream` as valid `body`?

No, we can also release the bug fix beforehand independently of the v4 update.

**Comment by jvpelt** [comment]
Hi @pi0, 

I’m running into something similar with a “tanstack start” app I’m trying to deploy to a SWA. Right now it returns a ReadableStream, but the azure functions v3 programming model doesn’t support that. A quick workaround could be to check the body type and turn the stream into a string. I can open a PR for that if you’d like.

On the other hand, v4 does support streams. I’ve seen a few issues about upgrading to v4, but I’m not really sure what the current status is.

**Comment by pi0** [comment]
If you can please feel free to open a PR to upgrade SWA in v3 branch.

**Comment by bishsbytes** [comment]
> Hi [@pi0](https://github.com/pi0),
> 
> I’m running into something similar with a “tanstack start” app I’m trying to deploy to a SWA. Right now it returns a ReadableStream, but the azure functions v3 programming model doesn’t support that. A quick workaround could be to check the body type and turn the stream into a string. I can open a PR for that if you’d like.
> 
> On the other hand, v4 does support streams. I’ve seen a few issues about upgrading to v4, but I’m not really sure what the current status is.

@jvpelt did you find a workaround for this? I'm hitting the same issue with Tanstack Start. I'm just getting a `{}` response.

**Comment by thomasschaller1** [comment]
Any news about this one?

**Comment by derkoe** [comment]
I'm using the workaround from @Xiang-CH from here https://github.com/Xiang-CH/tanstack-azure-swa-template  and it works without issues.

Can anyone have a look at their PR?
